### PR TITLE
Add support for unlimited file sizes

### PR DESCRIPTION
**Pull request by splatteredbits**

### CODE REVIEW

File: ProGetAutomation/Functions/Send-ProGetAsset.ps1 [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+                                       [IO.FileAccess]::Read,
+                                       [IO.FileShare]::Read,
+                                       4096,
+                                       [IO.FileOptions]::SequentialScan)
+
+    try
+    {
+        $fileLength = $fileStream.Length
+        $totalParts = [Math]::DivRem([long]$fileLength, [long]$MaxRequestSize, [ref]$remainder)
+        if($remainder -ne 0)
```
Comment (KhoiKy) [inline ProGetAutomation/Functions/Send-ProGetAsset.ps1:125]: Nit: Spacing inconsistent with other if/for statements in this file.

**Review by KhoiKy — DISMISSED** [review]

**Review by decoyjoe — APPROVED** [review]

File: Tests/Set-ProGetAsset.Tests.ps1 [hunk 2]
```diff
@@ -1,235 +1,228 @@
...
-        [string]
-        $Name,
-        [string]
-        $Directory
-    )
-    it ('should not contain the asset "{0}"' -f $Name) {
-        Get-ProGetAsset -Session $session -DirectoryName $baseDirectory -Path $Directory -ErrorAction Ignore | Where-Object { $_.name -match $name } | Should -BeNullOrEmpty
-    }
-}
+Describe 'Set-ProGetAsset' {
```
[the remark this answers is not in the document]
Comment (splatteredbits) [reply Tests/Set-ProGetAsset.Tests.ps1:129]: Hmm. I don't use the CodeLens to run tests. I run everything through the command line, e.g. `Invoke-Pester TEST -FullNameFilter FILTER`. That's *really* annoying the CodeLens won't run an individual `It` block. *sigh*

**Review by KhoiKy — DISMISSED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+# ProGetAutomation Changelog
+
+# 1.1.0
+
+## Added
+
+* Parameter `Url` to function `New-ProGetSession`. This replaces the `Uri` parameter. We'll always use a URL to connect
+to ProGet, since it's HTTP-based software.
+* Property `Url` on the ProGet session object. This replaces the `Uri` property.
+* Parameter `MaxRequestSize` to function `Set-ProGetAsset`. The `Set-ProGetAsset` function no uploads files in parts
```
Comment (decoyjoe) [inline CHANGELOG.md:11]: Typo: `no` should be `now`

File: ProGetAutomation/Functions/Send-ProGetAsset.ps1 [hunk 4]
```diff
@@ -0,0 +1,257 @@
+function Send-ProGetAsset
+{
+    <#
+    .SYNOPSIS
+    Uploads file to a ProGet asset directory. ***This is an internal ProGetAutomation function. Use `Set-ProGetAsset`
+    instead.
```
Comment (decoyjoe) [inline ProGetAutomation/Functions/Send-ProGetAsset.ps1:6]: Missing the trailing `***` characters.

**Review by decoyjoe — APPROVED** [review]

**Review by decoyjoe — APPROVED** [review]

**Pull request merged**